src/auth/integration-tests/src/lib.rs

```diff
@@ -267,6 +270,44 @@ pub async fn api_key() -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn mds_id_token() -> anyhow::Result<()> {
+    let audience = "https://example.com";
+
+    // Get the service account email from the metadata server directly
+    let client = reqwest::Client::new();
+    let expected_email = client
+        .get("http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/email")
+        .header("Metadata-Flavor", "Google")
+        .send()
+        .await
+        .expect("failed to get service account email from metadata server")
+        .text()
+        .await
+        .expect("failed to read service account email from metadata server response");
```

Comment:
Without a retry loop this may flake at any time. Can we add something simple, like trying 3 times?

Reply:
Feel free to do that in a followup PR.